src/typ.rs

```diff
@@ -1523,6 +1531,29 @@ impl<'a> Typer<'a> {
                 .map_err(|e| convert_unify_error(e, then.try_binding_location()))?;
         }
 
+        // Ensure exhaustiveness of value if this is a `let` binding
+        if kind == BindingKind::Let {
```

Comment:
We would want the same for `try` too, so `!= BindingKind::Assert` perhaps?

```rust
try "partial pattern" = Ok("no match")
```

Reply:
Wouldn't a try binding already have been type checked as being a result and be implicitly handling both the Ok and Error constructors?

Reply:
Yes but the contained value might not be matched, as with that string pattern in the example above

Reply:
Ah yes, got you.